payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardDetailsSectionElementUI.kt

```diff
@@ -0,0 +1,111 @@
+package com.stripe.android.ui.core.elements
+
+import android.content.Intent
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RestrictTo
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.stripe.android.stripecardscan.cardscan.CardScanSheetResult
+import com.stripe.android.stripecardscan.cardscan.exception.UnknownScanException
+import com.stripe.android.ui.core.DefaultIsStripeCardScanAvailable
+import com.stripe.android.ui.core.PaymentsTheme
+import com.stripe.android.ui.core.R
+import com.stripe.android.ui.core.cardscan.CardScanActivity
+
+@Composable
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun CardDetailsSectionElementUI(
+    enabled: Boolean,
+    controller: CardDetailsSectionController,
+    hiddenIdentifiers: List<IdentifierSpec>?
+) {
+    val context = LocalContext.current
+    val isStripeCardScanAvailable = DefaultIsStripeCardScanAvailable()
+
+    Row(
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = 9.dp),
+    ) {
+        H6Text(
+            text = stringResource(R.string.card_information),
+            modifier = Modifier
+                .semantics(mergeDescendants = true) { // Need to prevent form as focusable accessibility
+                    heading()
+                }
+        )
+        if (isStripeCardScanAvailable()) {
+            val cardScanLauncher =
```

Comment:
It might improve readability a little if this entire if statement is in a new composable: ScanCardButton?